### PR TITLE
Add Helm over Operator job to full E2E

### PR DESCRIPTION
**Pull request by dfarrell07**
As we move to better support Helm over Operator deployments, we need to
start adding tests to other repos. For now only add a basic,
default-only canary job.

Signed-off-by: Daniel Farrell <dfarrell@redhat.com>

### CODE REVIEW

**Comment by dfarrell07** [comment]
There are broken tests, commented out for now, that need fixes before this can go in.

**Comment by dfarrell07** [comment]
@vthapar I think you added these networkplugin-syncer tests in be79c3abafd1d334f4da1178b85e1b3402ff76d0 and service discovery tests in 69fd78e15ae8793183840d5e0519c61a313d3251 Any tips for getting them passing for Helm?

**Comment by mangelajo** [comment]
> @vthapar I think you added these networkplugin-syncer tests in [be79c3a](https://github.com/submariner-io/submariner-operator/commit/be79c3abafd1d334f4da1178b85e1b3402ff76d0) and service discovery tests in [69fd78e](https://github.com/submariner-io/submariner-operator/commit/69fd78e15ae8793183840d5e0519c61a313d3251) Any tips for getting them passing for Helm?

Probably those system accounts are missing in helm, I guess

**Comment by mangelajo** [comment]
> > @vthapar I think you added these networkplugin-syncer tests in [be79c3a](https://github.com/submariner-io/submariner-operator/commit/be79c3abafd1d334f4da1178b85e1b3402ff76d0) and service discovery tests in [69fd78e](https://github.com/submariner-io/submariner-operator/commit/69fd78e15ae8793183840d5e0519c61a313d3251) Any tips for getting them passing for Helm?
> 
> Probably those system accounts are missing in helm, I guess

and if that's the case, first good catch! ;)

File: scripts/kind-e2e/lib_subctl_gather_test.sh [hunk 1]
```diff
@@ -53,10 +53,12 @@ function validate_gathered_files () {
...
   validate_resource_files all 'endpointslices.discovery.k8s.io' 'EndpointSlice'
   validate_resource_files $subm_ns 'configmaps' 'ConfigMap' '-l component=submariner-lighthouse'
-  validate_resource_files kube-system 'configmaps' 'ConfigMap' '--field-selector metadata.name=coredns'
+  # FIXME: Broken with Helm
+  #validate_resource_files kube-system 'configmaps' 'ConfigMap' '--field-selector metadata.name=coredns'
 
   validate_pod_log_files $subm_ns '-l component=submariner-lighthouse'
-  validate_pod_log_files kube-system '-l k8s-app=kube-dns'
+  # FIXME: Broken with Helm
+  #validate_pod_log_files kube-system '-l k8s-app=kube-dns'
```
Comment (mangelajo) [inline scripts/kind-e2e/lib_subctl_gather_test.sh:61]: I wonder what's this one about

Comment (Jaanki) [reply scripts/kind-e2e/lib_subctl_gather_test.sh:61]: This comes from gather. This validates that a file with logs for pod with label `"k8s-app=kube-dns"` is generated.

File: .github/workflows/e2e-full.yml [hunk 2]
```diff
@@ -26,6 +27,7 @@ jobs:
         include:
           # This is the oldest K8s version we try to support
           - k8s_version: '1.17'
+          - deploytool: 'helm'
```
Comment (mangelajo) [inline .github/workflows/e2e-full.yml:30]: I would add another with globalnet too (as we have seen it to be problematic sometimes)

**Pull request closed without merging**